### PR TITLE
Require all owning documents to have permission policy enabled

### PR DESCRIPTION
**Pull request by kenchris**
@rakuco

***
<a href="https://pr-preview.s3.amazonaws.com/kenchris/compute-pressure/pull/239.html" title="Last updated on Oct 19, 2023, 2:47 PM UTC (fcec216)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/compute-pressure/239/3aa342e...kenchris:fcec216.html" title="Last updated on Oct 19, 2023, 2:47 PM UTC (fcec216)">Diff</a>

### CODE REVIEW

**Comment by kenchris** [comment]
Failure is due to respec.org being down!

File: index.html [hunk 1]
```diff
@@ -523,16 +523,21 @@ <h3>The <dfn>observe()</dfn> method</h3>
...
-          If |document| is not null and is not [=allowed to use=] the [=policy-controlled
-          feature=] token "compute-pressure", return [=a promise rejected with=] {{NotAllowedError}}.
+          [=list/For each=] |document:Document| in |relevantGlobal|'s [=owning document set=]:
+          <ol>
+            If |document| is not [=allowed to use=] the [=policy-controlled
+            feature=] token "compute-pressure", return [=a promise rejected with=] {{NotAllowedError}}.
+          </ol>
+        </li>
         <aside class="issue">
           <a href="https://github.com/wicg/compute-pressure/issues/110">
```
Comment (rakuco) [inline index.html:536]:
```suggestion
          <a href="https://github.com/w3c/compute-pressure/issues/110">
```

File: index.html [hunk 2]
```diff
@@ -523,16 +523,21 @@ <h3>The <dfn>observe()</dfn> method</h3>
...
-            Permission policy doesn't support workers yet #110
-          </a>
+          Permission policy doesn't support workers directly yet #110</a>, so they cannot be set per
```
Comment (rakuco) [inline index.html:537]:
```suggestion
          Permission policy does not support workers directly yet (#110)</a>, so they cannot be set per
```

**Review by rakuco — COMMENTED** [review]
The idea sounds good, I wonder if this spec is the first to implement this kind of check.

Can you mention this novel approach in the "Policy control" section? Maybe it's worth discussing it in "Mitigation strategies" too.

File: index.html [hunk 3]
```diff
@@ -283,6 +283,28 @@ <h3>
...
     Its [=policy-controlled feature/default allowlist=] is `["self"]`.
   </p>
+  <p>
+    Workers (dedicated and shared) adhere to the permission policy set by their
+    owning document(s).
+  </p>
+  <p>
+    Shared workers often have multiple owning documents as they can be obtained
+    by other documents with the [=same origin=].
+
```
Comment (rakuco) [inline index.html:293]: Did you mean to start another `<p>`aragraph here?

File: index.html [hunk 4]
```diff
@@ -283,6 +283,28 @@ <h3>
...
+    In this case, all owning documents have to have the permission policy allowed
```
Comment (rakuco) [inline index.html:294]: Is "have the permission policy allowed" a common way of phrasing this? It sounds a bit confusing.

File: index.html (hunk 3 above)
Comment (kenchris) [reply index.html:293]: No, just to make it easier for me to read in the editor

File: index.html (hunk 4 above)
Comment (kenchris) [reply index.html:294]: Yeah, I tried changing it a few times, but didn't find a good wording :-) Maybe @anssiko has some good suggestions

Comment (rakuco) [reply index.html:294]: How about just using [allowed to use](https://html.spec.whatwg.org/multipage/iframe-embed-object.html#allowed-to-use) like in `observe()`?

_"[...] all owning documents must be allowed to use the policy-controlled feature defined by this specification"_, for example.

Comment (kenchris) [reply index.html]: Sure, done

File: index.html [hunk 5]
```diff
@@ -283,6 +283,28 @@ <h3>
...
+  <p>
+    Workers (dedicated and shared) adhere to the permission policy set by their
+    owning document(s).
+  </p>
+  <p>
+    Shared workers often have multiple owning documents as they can be obtained
+    by other documents with the [=same origin=].
+
+    In this case, all owning documents must be [=allowed to use=] the policy-controlled
+    feature defined by this specification.
```
Comment (rakuco) [inline index.html:295]:
```suggestion
    In this case, all owning documents must be [=allowed to use=] the [=policy-controlled
    feature=] defined by this specification.
```

**Review by rakuco — APPROVED** [review]
lgtm with a nit

**Pull request merged**